text-generation-inference/server/text_generation_server/jetstream_pt_support/generator.py

```diff
@@ -477,6 +479,17 @@ def prefill(self, batch: Batch) -> Tuple[List[Generation], CachedBatch]:
         logger.debug("Model ready for decoding")
         return generations, batch
 
+    def _select_from_slots(self, logits: jnp.ndarray, batch_size: int=0) -> jnp.ndarray:
+        pad_token_id = self.tokenizer.pad_token_id
+        batch_size = logits.shape[0]
+        tokens = jnp.full((batch_size, 1), pad_token_id)
+        active_slots = [slot for slot in self.slots if slot.state == slot.State.READY]
+        for slot in active_slots:
```

Comment:
Maybe we can avoid the double iteration by filtering on the fly through the main iteration?

```suggestion
        for slot in filter(lambda slot: slot.state == slot.State.READY, slots):
```